veloroutes_voies_vertes/processing/covadis/export_covadis.py

```diff
@@ -0,0 +1,222 @@
+            }
+        )
+        self.addParameter(schema_param)
+
+        # Table à exporter
+        table_param = QgsProcessingParameterEnum(
+                self.TABLE,
+                tr("Donnée à exporter"),
+                options=self.EXPORTABLES,
+                defaultValue="",
+        )
+        self.addParameter(table_param)
+
+        # Nom du département pour le fichier d'export
+        depparam= QgsProcessingParameterString(
+            self.DPT,
+            tr("Département au format XXX"),
+            '066',
+            optional=False
+        )
+        self.addParameter(depparam)
+
+        outparam = QgsProcessingParameterFolderDestination(
+            self.PROJECTS_FOLDER,
+            tr("chemin de destination"),
+            defaultValue=False,
+            optional=False,
+            createByDefault=False)
+        self.addParameter(outparam)
+
+        self.addParameter(
+            QgsProcessingParameterBoolean(
+                self.CHARGER,
+                tr("Charger la couche correspondante dans le projet"),
+                defaultValue=False,
+                optional=False,
+            )
+        )
+
+        self.addOutput(
+            QgsProcessingOutputMultipleLayers(self.OUTPUT, tr("Couches de sortie"))
+        )
+
+    @staticmethod
+    def createExportTable(table, connection):
+        if "val" in table:
+            sql = "SELECT veloroutes.export_table('{}')".format(table)
+        else:
+            sql = "SELECT veloroutes.export_{}()".format(table)
+        try:
+            connection.executeSql(sql)
+        except QgsProviderConnectionException as e:
+            msg = e.args[0]
+            raise QgsProcessingException(msg)
+
+    def toSHP(self, context, table, dpt, dirname, uri, geom, charger):
+        uri.setDataSource('exports', table, geom, "")
+        layer = QgsVectorLayer(uri.uri(), table, "postgres")
+        if not layer.isValid():
+            return False
+
+        # Construction filename
+        prefixe="N_3V_"
+        suffixe = "_" + dpt
+        tablename= table.upper()
+        if table == 'element':
+            tablename= 'R_'+tablename +'_PORTION'
+        if 'poi' in table and 'portion' not in table:
+            tablename= tablename[4:]
+        if "val" in table:
+            prefixe="3V_"
+            suffixe=""
+            if "avancement" in table:
+                tablename = "AVANCEMENT_VAL"
+            if "statut" in table:
+                tablename = "STATUT_VAL"
+            if "portion" in table or "repere" in table:
+                tablename="TYPE"+tablename
+        geomtype = {
+            QgsWkbTypes.LineGeometry: '_L',
+            QgsWkbTypes.PointGeometry: '_P',
+            QgsWkbTypes.NullGeometry: ''
+        }
+        geomcode = geomtype[layer.geometryType()]
+        filename= prefixe +tablename + geomcode + suffixe
+
+        # Enregistrement du fichier shape
+        error = QgsVectorFileWriter.writeAsVectorFormat(
```

Comment:
```suggestion
        error = QgsVectorFileWriter.writeAsVectorFormatV2(
```
La fonction d'avant est dépréciée. Il faut utiliser la V2, il y peut-être un ajustement dans la signature.